packages/core/test/types.test.ts

```diff
@@ -34,59 +35,122 @@ describe('StateSchema', () => {
     elapsed: number;
   }
 
-  const lightMachine = Machine<LightContext, LightStateSchema, LightEvent>({
-    key: 'light',
-    initial: 'green',
-    meta: { interval: 1000 },
-    context: { elapsed: 0 },
-    states: {
-      green: {
-        id: 'green',
-        meta: { name: 'greenLight' },
-        on: {
-          TIMER: 'yellow',
-          POWER_OUTAGE: 'red'
-        }
-      },
-      yellow: {
-        on: {
-          TIMER: 'red',
-          POWER_OUTAGE: 'red'
-        }
-      },
-      red: {
-        on: {
-          TIMER: 'green',
-          POWER_OUTAGE: 'red'
+  it('should work with a StateSchema defined', () => {
+    const lightMachine = Machine<LightContext, LightStateSchema, LightEvent>({
+      key: 'light',
+      initial: 'green',
+      meta: { interval: 1000 },
+      context: { elapsed: 0 },
+      states: {
+        green: {
+          id: 'green',
+          meta: { name: 'greenLight' },
+          on: {
+            TIMER: 'yellow',
+            POWER_OUTAGE: 'red'
+          }
         },
-        initial: 'walk',
-        states: {
-          walk: {
-            on: {
-              PED_COUNTDOWN: 'wait'
-            }
+        yellow: {
+          on: {
+            TIMER: 'red',
+            POWER_OUTAGE: 'red'
+          }
+        },
+        red: {
+          on: {
+            TIMER: 'green',
+            POWER_OUTAGE: 'red'
           },
-          wait: {
-            on: {
-              PED_COUNTDOWN: {
-                target: 'stop',
-                cond: (ctx, e: { type: 'PED_COUNTDOWN'; duration: number }) => {
-                  return e.duration === 0 && ctx.elapsed > 0;
+          initial: 'walk',
+          states: {
+            walk: {
+              on: {
+                PED_COUNTDOWN: 'wait'
+              }
+            },
+            wait: {
+              on: {
+                PED_COUNTDOWN: {
+                  target: 'stop',
+                  cond: (
+                    ctx,
+                    e: { type: 'PED_COUNTDOWN'; duration: number }
+                  ) => {
+                    return e.duration === 0 && ctx.elapsed > 0;
+                  }
                 }
               }
+            },
+            stop: {
+              always: { target: '#green' }
             }
-          },
-          stop: {
-            always: { target: '#green' }
           }
         }
       }
-    }
+    });
+
+    noop(lightMachine);
+    expect(true).toBeTruthy();
   });
 
-  noop(lightMachine);
+  it('should work with a StateSchema defined inline', () => {
```

Comment:
This test only tests the event schema so maybe a more precise title would be better here?
```suggestion
  it('should work with an event schema defined inline', () => {
```